### PR TITLE
Cleanup NASSP import

### PR DESCRIPTION
**Pull request by aaronschachter**
### What's this PR do?

Now that we have a [final source file to import](https://dosomething.slack.com/archives/CTVPG6L4R/p1593459055107200?thread_ts=1593452429.105500&cid=CTVPG6L4R) for the NASSP groups, this pull request makes a few tweaks to the `rogue:nassp-groups-import` console command added in #1056:

* Create an additional new group type, `National Association of Secondary School Principals`, which will contain each unique school/name/city row within the file. For this reason, we no longer need to skip a row if it doesn't have any values set in the `NHS`, `NJHS`, or `NASC` columns.

* Modifies column list per our final import file. I've uploaded the file to a [gist](https://gist.githubusercontent.com/aaronschachter/9cd40a703a43e22077e78e12e011a7de/raw/1d689400d64f461d51dce0e99d2b16334c36b10e/nassp.csv) and have been testing the import using that URL.

* Rename the `external_id` field to `school_id`, and save the `universalid` column that's been added on our end to the import file. Refs [Slack thread](https://dosomething.slack.com/archives/CTVPG6L4R/p1593462442109800?thread_ts=1593452429.105500&cid=CTVPG6L4R)

    * Change the Groups transformer and create/edit forms accordingly

 
### How should this be reviewed?

👀 

### Any background context you want to provide?

I plan to add documentation to start detailing some of these import commands that we may still use (and perhaps it's time to delete ones we know we'll never again use), but holding off on adding to this PR to help keep it reviewable.

### Relevant tickets

References [Pivotal #173408737](https://www.pivotaltracker.com/story/show/173408737).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [x] Added appropriate feature/unit tests.

### CODE REVIEW

File: app/Console/Commands/ImportNasspGroupsCommand.php [hunk 1]
```diff
@@ -86,44 +86,61 @@ public function handle()
...
+        ]);
 
         info('rogue:nassp-groups-import: Beginning import...');
 
         /**
          * Without specifying these column names, this command fails with error:
          * "The header record must be an empty or a flat array with unique string value."
          */
-        $records = $csv->getRecords(['LABEL NAME', 'NHS', 'NJHS', 'NASC', 'NASSP', 'ADDRESS1', 'ADDRESS2', 'CITY', 'STATE', 'POSTAL', 'COUNTRY', 'NCES']);
+        $records = $csv->getRecords(['name', 'NHS', 'NJHS', 'NASC', 'NASSP', 'city', 'state', 'zip', 'universalid', 'ncescode', 'gsid']);
```
Comment (weerd) [inline app/Console/Commands/ImportNasspGroupsCommand.php:100]: Just curious if there's a reason for the column name variety in capitalization and why not the same across the different columns?

File: docs/endpoints/groups.md [hunk 2]
```diff
@@ -48,9 +48,9 @@ Example Response:
       "id": 23,
       "group_type_id": 1,
       "name": "Afton Central School",
-      "external_id": null,
       "city": "Afton",
       "state": "NY",
+      "external_id": null,
```
Comment (weerd) [inline docs/endpoints/groups.md:53]: should this be `"school_id": null,`?

File: tests/Console/ImportNasspGroupsCommandTest.php [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+use Rogue\Models\GroupType;
+
+class ImportNasspGroupsCommandTest extends TestCase
+{
+    public function testImportingGroups()
+    {
+        $this->artisan('rogue:nassp-groups-import', ['path' => 'tests/Console/example-nassp-groups.csv']);
+
+        // Four group types should have been created.
+        $this->assertTrue(GroupType::all()->count() == 4);
```
Comment (weerd) [inline tests/Console/ImportNasspGroupsCommandTest.php:16]: Small nitpick but can we switch to using `===` on the asserts to check against the count?

File: tests/Console/ImportNasspGroupsCommandTest.php [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            'city' => 'Scio',
+            'state' => 'NY',
+            'school_id' => '3603585',
+        ]);
+        $this->assertDatabaseMissing('groups', [
+            'name' => $groupName,
+            'group_type_id' => $nascGroupTypeId,
+        ]);
+
+        // Spot check Donald Duck was not imported for any groups besides NASSP.
```
Comment (weerd) [inline tests/Console/ImportNasspGroupsCommandTest.php:97]: 🦆

File: app/Console/Commands/ImportNasspGroupsCommand.php (hunk 1 above)
Comment (aaronschachter) [reply app/Console/Commands/ImportNasspGroupsCommand.php]: The uppercase was leftover from the initial file used in #1056. I updated the new file to have all lowercase and made corresponding changes in https://github.com/DoSomething/rogue/pull/1059/commits/8789aef509c8412157c73616c509d27bc4dd7e23. Thanks for the thorough review!

**Pull request merged**